src/engine/FileManager.ts

```diff
@@ -39,26 +39,27 @@ class FileManager {
      * @param excludes See config settings: a list of patterns to exclude
      */
     public getFiles(sources: ISourceEntry[], includes: string[], excludes: string[]): ISourceEntry[] {
-        const filesToCopy = new Array<ISourceEntry>()
-            , noneFound = new Array<string>();
+        const filesToCopy = new Array<ISourceEntry>(),
+            noneFound = new Array<string>();
 
-        const isEntryMatch = (entry: string, fileName: string) => fileName === entry ||
-            entry.startsWith('*') && fileName.endsWith(entry.slice(1)) ||
-            entry.endsWith('*') && fileName.startsWith(entry.slice(0, -1));
+        const isEntryMatch = (entry: string, fileName: string) =>
+            fileName === entry ||
+            (entry.startsWith('*') && fileName.endsWith(entry.slice(1))) ||
+            (entry.endsWith('*') && fileName.startsWith(entry.slice(0, -1)));
 
         for (let src of sources) {
-            const files = fs.readdirSync(src.path);
+            const files = this.getRecursiveListOfClsFiles(src.path, []); //fs.readdirSync(src.path);
 
-            if (files && files.some(file => file.endsWith('cls'))) {
-                files.forEach(fileName => {
+            if (files && files.some((file) => file.endsWith('cls'))) {
+                files.forEach((fileName) => {
```

Comment:
I wonder if we should move this filtering for `cls` files up into your new utility method so that it will truly do as the name suggests (it currently returns a list of all files), then we can remove everything after `&&` on line 53 and the condition currently on lines 56-58. What do you think?